### PR TITLE
refactor: using require instead of revert where possible

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

closes: #7

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [x] Wrote unit and integration tests.
- [x] Added relevant natspec and `godoc` comments.
- [x] Provide a [conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) to follow the repository standards.
- [x] Re-reviewed `Files changed` in the GitHub PR explorer.
- [x] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: src/ICS26Router.sol [hunk 1]
```diff
@@ -59,12 +57,8 @@ contract ICS26Router is IICS26Router, IICS26RouterErrors, Ownable, ReentrancyGua
             newPortId = Strings.toHexString(app);
         }
 
-        if (apps[newPortId] != IIBCApp(address(0))) {
-            revert IBCPortAlreadyExists(newPortId);
-        }
-        if (!IBCIdentifiers.validatePortIdentifier(bytes(newPortId))) {
-            revert IBCInvalidPortIdentifier(newPortId);
-        }
+        require(apps[newPortId] == IIBCApp(address(0)), IBCPortAlreadyExists(newPortId));
```
Comment (srdtrk) [inline src/ICS26Router.sol:60]:
```suggestion
        require(address(apps[newPortId]) == address(0), IBCPortAlreadyExists(newPortId));
```

File: src/utils/IBCStore.sol [hunk 2]
```diff
@@ -35,26 +35,26 @@ contract IBCStore is IIBCStore, IICS24HostErrors, Ownable {
...
                 ICS24Host.packetCommitmentPathCalldata(packet.sourceChannel, packet.sequence)
-            );
-        }
+            )
+        );
 
         bytes32 commitment = ICS24Host.packetCommitmentBytes32(packet);
-        emit PacketCommitted(path, commitment);
         commitments[path] = commitment;
+        emit PacketCommitted(path, commitment);
```
Comment (srdtrk) [inline src/utils/IBCStore.sol:47]: makes more logical sense for this to be at the end. This event should probably be removed on another issue to clean up unnecessary events

Comment (gjermundgaraba) [reply src/utils/IBCStore.sol:47]: Yes, I added an issue for implementing spec events, and specified in that issue that we should clean up events all over the place at the same time: https://github.com/cosmos/solidity-ibc-eureka/issues/108

File: .solhint.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "named-parameters-mapping": "warn",
     "no-console": "off",
     "not-rely-on-time": "off",
-    "one-contract-per-file": "off"
+    "one-contract-per-file": "off",
+    "gas-custom-errors": "off"
```
Comment (gjermundgaraba) [inline .solhint.json:13]: I understand why this one is necessary right now, but I created an issue to re-introduce this one as soon as this is fixed in solhint: https://github.com/cosmos/solidity-ibc-eureka/issues/109

**Review by gjermundgaraba — APPROVED** [review]
Very nice! I very much like this syntax better. Mostly reads a lot better, which I love :D

LGTM!

**Pull request merged**